ReducingCrime.Rmd

```diff
@@ -23,56 +28,62 @@ Then we load the dataset, which is in the same directory as this RMD.
 rawCrimeData = read.csv("crime_v2.csv")
 str(rawCrimeData)
 ```
-The dataset contains 25 columns (variables) and 97 rows
+The dataset contains **25** variables and **97** records. Now, we'll need to do few rounds of data cleanup before analyzing any further.
 
 ## Data Quality/Clean-up
 
-### Remove NAs
-```{r fig.align='center', fig.height=3, fig.width=3, fig.show='hold'}
-sum(is.na(rawCrimeData$county))
-```
-The data set contains 6 rows with NA county. We need to remove these rows before proceeding further.
-```{r fig.align='center', fig.height=3, fig.width=3, fig.show='hold'}
-crimeData <- rawCrimeData[!is.na(rawCrimeData$county),]
-min(complete.cases(crimeData))
-```
-Since there are no more incomplete rows, we're good to go.
-
 ### Convert county to factor
 Since county is not a measurement, we should convert it into factor so that it can be easily used for further analysis.
 
 ```{r fig.align='center', fig.height=3, fig.width=3, fig.show='hold'}
-crimeData$county <- as.factor(crimeData$county)
-length(levels(crimeData$county))
+rawCrimeData$county <- as.factor(rawCrimeData$county)
+length(levels(rawCrimeData$county))
 ```
-This is interesting, we have 91 rows but only 90 levels.
-Eyeballing the data shows there are two identical rows for county 193, same can be varifired using duplicated function.
-```{r fig.align='center', fig.height=3, fig.width=3, fig.show='hold'}
-crimeData[duplicated(crimeData),"county"]
-```
-so lets delete the duplicate row
+
+Interestingly we have 91 rows but only 90 levels. Eyeballing the data shows there are two identical rows for county 193, same can be verified using duplicated function.
 ```{r fig.align='center', fig.height=3, fig.width=3, fig.show='hold'}
-crimeData <- crimeData[!duplicated(crimeData),]
+rawCrimeData[duplicated(rawCrimeData),"county"]
+
+#so lets delete the duplicate row
+#before..
+nrow(rawCrimeData)
+rawCrimeData <- rawCrimeData[!duplicated(rawCrimeData),]
+#after..
+nrow(rawCrimeData)
 ```
 
 ### Convert prbconv to number
 Now lets convert prbconv from factor to number because it is a probability value.
 ```{r fig.align='center', fig.height=3, fig.width=3, fig.show='hold'}
-crimeData$prbconv <- as.numeric(levels(crimeData$prbconv))[crimeData$prbconv]
-summary(crimeData$prbconv)
+rawCrimeData$prbconv <- as.numeric(levels(rawCrimeData$prbconv))[rawCrimeData$prbconv]
+summary(rawCrimeData$prbconv)
+```
+
+### Remove NAs
+```{r fig.align='center', fig.height=3, fig.width=3, fig.show='hold'}
+#let us find how many records we have first..
+sum(is.na(rawCrimeData$crmrte))
+sum(is.na(rawCrimeData$county))
+```
+
+```{r fig.align='center', fig.height=3, fig.width=3, fig.show='hold'}
+#The data set containsrows with NA county and NA crime rate. 
+rawCrimeData <- rawCrimeData[!is.na(rawCrimeData$crmrte),]
+crimeData <- rawCrimeData[!is.na(rawCrimeData$county),]
+min(complete.cases(crimeData))
 ```
 
 ### Ceil probabilities to 1 to fit them in valid probability range
```

Comment:
After reading through the Lab, these are proxy to actual rations and not actual rations. So I believe it is ok to allow > 1?